### PR TITLE
Set color of tab icon bottom navigation view

### LINKED ISSUE #912: Background Color of Tab Icon Should Adjust According to Primary Colour
**Actual Behaviour**

The background colour for highlighting the current tab is fixed. So if the primary colour is same as the highlight colour the tab icon becomes invisible. 

**Expected Behaviour**
The background colour of the icon should adjust according to primary colour. 

**Steps to reproduce it**
Switch primary colour to dark grey
**Screenshots of the issue**
image

**Would you like to work on the issue?**
Yes

### PR DESCRIPTION
**Pull request by mayank8318**
Fix #912 

Changes: Make a ColorStateList and set the color for checked and unchecked states dynamically.

Screenshots for the change: 
image
image

### CODE REVIEW

**Comment by mohitmanuja** [comment]
@mayank408 The purpose is not only to make the home button visible but also needs to be uniform. So try to make it uniform with all other icons.

**Comment by vinaysajjanapu** [comment]
rebase and squash your commits

**Comment by mayank8318** [comment]
@vinaysajjanapu If the primary colour is black then the unchecked icons wont be visible so I am using light gray which looks good.

File: app/src/main/java/org/fossasia/phimpme/base/BaseActivity.java [hunk 1]
```diff
@@ -74,12 +89,25 @@ void selectBottomNavigationBarItem(int itemId) {
         }
     }
 
+    void setIconColor(int colorToInvert){
+        if(Color.red(colorToInvert) < 60 && Color.green(colorToInvert) < 60 && Color.blue(colorToInvert) < 60){
+            colors[0] = Color.WHITE;
+            colors[1] = Color.GRAY;
```
Comment (vinaysajjanapu) [inline app/src/main/java/org/fossasia/phimpme/base/BaseActivity.java]: change your function with

    void setIconColor(int color){
        if(Color.red(color) + Color.green(color)+ Color.blue(color) < 300)
            colors[0] = Color.WHITE;
        else
            colors[0] = Color.BLACK;
    }

File: app/src/main/java/org/fossasia/phimpme/base/BaseActivity.java [hunk 2]
```diff
@@ -18,11 +20,24 @@
...
     protected BottomNavigationView navigationView;
 
+    private int[][] states = new int[][] {
+            new int[] {android.R.attr.state_checked}, // checked
+            new int[] {-android.R.attr.state_checked}, // unchecked
+    };
+
+    private int[] colors = new int[] {
+            Color.BLACK,
+            Color.WHITE,
```
Comment (vinaysajjanapu) [inline app/src/main/java/org/fossasia/phimpme/base/BaseActivity.java]: Change second value in colors array to 
Color.parseColor("#70ffffff")

**Comment by vinaysajjanapu** [comment]
@mayank8318 please make the requested changes
Update the screenshots, so that it will be easier for other members to review.
Please rebase and squash properly

**Comment by mayank8318** [comment]
@vinaysajjanapu Opened a new PR #964 with all the changes as my commits were not being synced.
So I should close this right?

**Pull request closed without merging**